packages/fxa-auth-client/lib/client.ts

```diff
@@ -1433,6 +1433,131 @@ export default class AuthClient {
     return response;
   }
 
+  /**
+   * Change password using JWT authentication (for MFA-protected operations).
+   * This uses the new /mfa/password/change endpoint that requires JWT authentication.
+   *
+   * @param jwt - The JWT token for authentication.
+   * @param email - The email of the user.
+   * @param oldPassword - The old password of the user.
+   * @param newPassword - The new password of the user.
+   * @param sessionToken - The session token of the user.
+   * @param options - The options for the password change.
+   * @param headers - The headers for the request.
+   */
+  async passwordChangeWithJWT(
+    jwt: string,
+    email: string,
+    oldPassword: string,
+    newPassword: string,
+    sessionToken: string,
+    options: {
+      keys?: boolean;
+      skipCaseError?: boolean;
+      reauthEmail?: string;
+    } = {},
+    headers?: Headers
+  ): Promise<SignedInAccountData> {
+
+    console.log('passwordChangeWithJWT', email, oldPassword, newPassword, sessionToken, options, headers);
+
+    const oldCredentials = await this.sessionReauth(sessionToken, options.reauthEmail || email, oldPassword, {
+      keys: true
+    }, headers);
+    const oldCredentialsAuth = await crypto.getCredentials(
+      email,
+      oldPassword
+    );
+    const oldAuthPW = oldCredentialsAuth.authPW;
+
+    const keys = await this.accountKeys(
+      oldCredentials.keyFetchToken!,
+      oldCredentials.unwrapBKey!,
+      headers
+    );
+
+    const newCredentials = await crypto.getCredentials(
+      email,
+      newPassword
+    );
+
+    const wrapKb = crypto.unwrapKB(keys.kB, newCredentials.unwrapBKey);
+    const authPW = newCredentials.authPW;
+
+    let payload: {
+      email: string;
+      oldAuthPW: string;
+      authPW: string;
+      wrapKb: string;
+      authPWVersion2?: string;
+      wrapKbVersion2?: string;
+      clientSalt?: string;
+    } = {
+      email,
+      oldAuthPW,
+      authPW,
+      wrapKb,
+    };
+    if (this.keyStretchVersion === 2) {
+      const status = await this.getCredentialStatusV2(email, headers);
```

Comment:
Should we get the v2 credentials at the start of the routine? That way we could send them to `sessionReauth` (we would need to support this). Using v2 credentials has the advantage of never resulting in an `INCORRECT_EMAIL_CASE` which will adds an extra roundtrip.